File: src/main/java/nl/rabobank/assessment/persistence/entity/Person.java
package nl.rabobank.assessment.persistence.entity;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.domain.Persistable;

import java.time.LocalDate;
import java.util.Objects;
import java.util.UUID;

@Getter
@Setter
@ToString
public class Person implements Persistable<UUID> {

	@Id
	@Setter(AccessLevel.NONE)
	private UUID id;

	private String firstName;

	private String lastName;

	private LocalDate dateOfBirth;

	private String street;

	private Integer houseNumber;

	private String houseNumberAdditions;

	private String postalCode;

	private String city;

	private String country;

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object other) {
		if (this == other)
			return true;
		if (other == null || getClass() != other.getClass())
			return false;
		Person that = (Person) other;
		return Objects.equals(id, that.id);
	}

	@Override
	public boolean isNew() {
		if (id == null) {
			id = UUID.randomUUID();
			return true;
		}
		return false;
	}
}

File: src/main/kotlin/nl/rabobank/assessment/persistence/entity/Person.kt
package nl.rabobank.assessment.persistence.entity

import org.springframework.data.annotation.Id
import org.springframework.data.domain.Persistable
import java.time.LocalDate
import java.util.*

data class Person(

    @Id
    private var id: UUID? = null,
    var firstName: String,
    var lastName: String,
    var dateOfBirth: LocalDate,
    var street: String,
    var houseNumber: Int,
    var houseNumberAdditions: String?,
    var postalCode: String,
    var city: String,
    var country: String

) : Persistable<UUID> {

    override fun getId(): UUID? {
        return id;
    }

    override fun isNew(): Boolean {
        if (id == null) {
            id = UUID.randomUUID()
            return true
        }
        return false
    }
}